### ISSUE TITLE
E 0:00:01:0253   MainLoop.gd:57 @ _on_save_button_pressed(): Condition "!is_ancestor_of(p_node)" is true. Returning: false   <C++ Source>   scene/main/node.cpp:2302 @ is_editable_instance()   <Stack Trace>  MainLoop.gd:57 @ _on_save_button_pressed()

### ISSUE THREAD

**Issue by doopt5** [godotengine/godot#81982]
### Godot version

Godot_V4.1.1 Stable

### System information

Pop OS, Intel(R) Core(TM) i5-1035G1 , Intel UHD Graphics 620, Driver Version 31.0.101.211, Vulkan 1.3.246

### Issue description

On this code:
![image](https://github.com/godotengine/godot/assets/53439238/460dd2c8-1fae-417a-8a6a-0963c48be035)

I am receiving this error:
![image](https://github.com/godotengine/godot/assets/53439238/4b37af8d-0498-4cc6-825a-e556e7dd523f)

Scene tree:
![image](https://github.com/godotengine/godot/assets/53439238/4e1a9334-a3fc-457b-a4e2-c5792e7525ad)

### Steps to reproduce

Trying to pack a node and its children in a scene:
![image](https://github.com/godotengine/godot/assets/53439238/9872825b-8dff-4221-a15b-04a53fc37e46)

With this code:
![image](https://github.com/godotengine/godot/assets/53439238/38bce8cd-a50f-4ddd-a94d-dc09efde3a42)
 

### Minimal reproduction project

[ANT-ItemCreator.zip](https://github.com/godotengine/godot/files/12673331/ANT-ItemCreator.zip)

**Comment by SomeSortOfSam** [comment]
Actual minimum reproduction: 
[SaveBugn't.zip](https://github.com/godotengine/godot/files/12717851/SaveBugn.t.zip)
This isn't a bug, but it is weird that you can't add a node to packed scene while it's in the scene tree. Does this have something to do with the node itself holding parentage information?
Regardless, this is easily worked around by doing `packed_scene.pack(save_file.duplicate())`

**Comment by stickgrinder** [comment]
Hi. This is indeed happening to me too and the workaround proposed by @SomeSortOfSam doesn't seem to help.

If I understood correctly, the `duplicate()` is there to have a copycat of the node, out of the scene tree, right?

**Comment by doopt5** [comment]
> Hi. This is indeed happening to me too and the workaround proposed by @SomeSortOfSam doesn't seem to help.
> 
> If I understood correctly, the `duplicate()` is there to have a copycat of the node, out of the scene tree, right?

Yes I was unable to get it to work with duplicate(). I am confused because this worked in 4.0 when it came out. This also worked in 3.4.3 which is the last version I was using of 3.0.

**Comment by doopt5** [comment]
_Disregard close. Accidental click._

**Comment by stickgrinder** [comment]
> Yes I was unable to get it to work with duplicate(). I am confused because this worked in 4.0 when it came out. This also worked in 3.4.3 which is the last version I was using of 3.0.

duplicate() isn't working for me neither. I think I stumbled onto another thread some days ago, talking about a regression in Godot 4.0 that was fixed so maybe this is still the case?

If I can find it again, maybe it can be applied to 4.1 or give some clue...

**Comment by stickgrinder** [comment]
Today I worked around this error with a stupid trick. It applies to my case but this may help something else, so I'm leaving this here.

Before getting into the details: I'm pretty sure this error is a regression but I can't find the thread that talked about this. 

This error is not actually hindering the ability to pack the element, but it complains about wrong ownership (at least in my case).

The starting situation is this: I have those "Props" elements which are instance of scenes already saved externally. Some of those nodes have a polygon attached to react to clicks.

![image](https://github.com/godotengine/godot/assets/625436/2f916c8d-4193-4838-a286-c8c3101416b8)

The polygons are children of the nodes I want to save but owned by the root scene. The packer ignores them but triggers this ugly error.

My solution is to remove the offending children on the fly and reassigning them to the proper parent afterwards. The following code is in a loop (on `Props.get_children()`) and it account a bit for other possible children that I may want to add in the future.

```
	var packed_scene: PackedScene = PackedScene.new()

	# Workaround starts here #################
	var not_owned_children = []
	for child in prop.get_children():
		if child.owner == prop: continue    # do nothing if the child is owned by its parent

		not_owned_children.append(child)     # add the not-owned child to a list
		child.get_parent().remove_child(child)   # Unparent it

	packed_scene.pack(prop)   # pack the node that's now clean
	ResourceSaver.save(packed_scene, prop.scene_file_path)

	# Workaround restarts here
	for child in not_owned_children:
		prop.add_child(child)    # Reassign the child where it belongs
```

This does the trick for me, but I will do more research in the next days because this is terrible and I'm sorry with my mom.

Hope this may help someone else in the meantime.

**Comment by Roboticy3** [comment]
I have a *very* scuffed save/load concept I am trying to hack out at 1:30am. Mistakes were made here and I don't think I'm in a position to blame the engine devs. But I ran into this same issue. I have a manually instantiated node being added as a child to another node, the ```owner``` of the node is ```null```. Nodes that are added under this parent in the scene editor (e.g. before runtime) work fine, it's just the nodes added at runtime that have this issue. And so, inevitably, the ```pack``` function throws and error and gives up. Luckily these are all direct children, so this workaround should get it done. I'll edit my comment when I've tested it.
